acceptance-tests/access_control_test.go

```diff
@@ -0,0 +1,104 @@
+package acceptance_tests
+
+import (
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+/*
+	Test strategy:
+		* Use an SSH tunnel to make requests to HAProxy that appear to come from 127.0.0.1
+		* Requests directly from test runner on Concourse appear to come from 10.0.0.0/8
+		We can test whitelisting and blacklisting by using these CIDRs
+*/
+var _ = Describe("Access Control", func() {
+	opsfileWhitelist := `---
+# Enable CIDR whitelist
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/cidr_whitelist?
+  value: ((cidr_whitelist))
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/block_all?
+  value: true
+`
+	opsfileBlacklist := `---
+# Enable CIDR blacklist
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/cidr_blacklist?
+  value: ((cidr_blacklist))
+`
+
+	It("Allows IPs in whitelisted CIDRS", func() {
+		deploymentName := "haproxy"
+		haproxyBackendPort := 12000
+
+		// Allow 127.0.0.1/32, but deny other connections
+		haproxyInfo, _ := deployHAProxy(baseManifestVars{
+			haproxyBackendPort:    12000,
+			haproxyBackendServers: []string{"127.0.0.1"},
+			deploymentName:        deploymentName,
+		}, []string{opsfileWhitelist}, map[string]interface{}{
+			"cidr_whitelist": []string{"127.0.0.1/32"},
+		}, true)
+		defer deleteDeployment(deploymentName)
+
+		closeLocalServer, localPort := startDefaultTestServer()
+		defer closeLocalServer()
+
+		closeBackendTunnel := setupTunnelFromHaproxyToTestServer(haproxyInfo, haproxyBackendPort, localPort)
+		defer closeBackendTunnel()
+
+		// Set up a tunnel so that requests to localhost:11000 appear to come from 127.0.0.1
+		// on the HAProxy VM as this is now whitelisted
+		closeFrontendTunnel := setupTunnelFromLocalMachineToHAProxy(haproxyInfo, 11000, 80)
+		defer closeFrontendTunnel()
+
+		By("Denying access from non-whitelisted CIDRs (request from test runner)")
+		_, err := http.Get(fmt.Sprintf("http://%s", haproxyInfo.PublicIP))
+		Expect(err.Error()).To(ContainSubstring("EOF")) // when denying HAProxy returns an empty response
```

Comment:
```
Expect(err).To(Equal(io.EOF))
```
?

Reply:
This is actually a url.Error that looks like this:

```
 <*url.Error | 0xc0004281e0>: {
          Op: "Get",
          URL: "http://10.245.0.11",
          Err: <*errors.errorString | 0xc000060100>{s: "EOF"},
      }
```

I've improved the strictness by changing the check to 

```
_, err := http.Get(fmt.Sprintf("http://%s", haproxyInfo.PublicIP))
Expect(err).To(HaveOccurred())
urlErr, _ := err.(*url.Error)
Expect(urlErr.Unwrap()).To(Equal(io.EOF))
```